### PR TITLE
babel-plugin-expo-vector-icons

### PR DESCRIPTION
**Pull request by EvanBacon**
# How

Removed the re-export of vector-icons. You will now have to import your fonts like so

```diff
- import { Icons } from 'expo';

+ import { Ionicons } from '@expo/vector-icons';
```

Which will then become a direct import after bundling:

```diff
- import { Ionicons } from '@expo/vector-icons';

+ import Ionicons from '@expo/vector-icons/build/Ionicons';
// Same as doing `import Ionicons from '@expo/vector-icons/Ionicons';` but with one less file.

```

> This applies to native as well.

# Test Plan

In the future, running `expo build:web` will generate a project with only the font files you use and none of the extra fonts. Currently you get all of these bundled even if you never use fonts:

```
AntDesign.ttf
Entypo.ttf
Feather.ttf
FontAwesome.ttf
FontAwesome5_Brands.ttf
FontAwesome5_Solid.ttf
Foundation.ttf
Ionicons.ttf
MaterialCommunityIcons.ttf
MaterialIcons.ttf
SimpleLineIcons.ttf
```

### CODE REVIEW

**Review by sjchmiela — APPROVED** [review]
Apart from these two comments I would say it looks good to me. Are we looking to merge it?

Isn't importing icons from `expo` deprecated for a while now so we could remove it altogether and remove the root need for this package?

**Comment by brentvatne** [comment]
> Isn't importing icons from `expo` deprecated for a while now so we could remove it altogether and remove the root need for this package?

since it was removed we have added it back as a global because it's still depended on in the ecosystem: https://github.com/expo/expo/blob/master/packages/expo/src/globals.ts

**Comment by brentvatne** [comment]
i think we should add this to the vector-icons repo so it's clear that changes to @expo/vector-icons may require updating this plugin

i like this pr though!

**Review by brentvatne — APPROVED** [review]

File: packages/babel-plugin-expo-vector-icons/package.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "publishConfig": {
+    "registry": "https://registry.npmjs.org/"
+  },
+  "name": "babel-plugin-expo-vector-icons",
+  "version": "1.0.0",
+  "description": "Babel plugin for eliminating unused fonts by redirecting all @expo/vector-icons",
+  "main": "index.js",
+  "devDependencies": {
+    "babel-core": "^6.26.3",
```
Comment (ide) [inline packages/babel-plugin-expo-vector-icons/package.json]: Having two versions of Babel in the repo is more likely to cause issues somewhere down the line. Update this to use Babel 7.

**Review by ide — CHANGES_REQUESTED** [review]
Looks good except for the Babel 6 usage. Also we need to have a smooth transition plan for deprecating `import { Icons } from 'expo';`.

**Comment by EvanBacon** [comment]
stale, closing. We may investigate using SVG icons on web instead.

**Pull request closed without merging**